Workout Core/Model/Workout/Workout.swift

```diff
@@ -519,4 +520,169 @@ public class Workout {
 		}
 	}
 	
+    private let typeRow = 0
+    private let startRow = 1
+    private let endRow = 2
+    private let durationRow = 3
+    private var distanceRow: Int? {
+        guard self.totalDistance != nil else {
+            return nil
+        }
+        
+        return 1 + durationRow
+    }
+    private var avgHeartRow: Int? {
+        guard self.avgHeart != nil else {
+            return nil
+        }
+        
+        return 1 + (distanceRow ?? durationRow)
+    }
+    private var maxHeartRow: Int? {
+        guard self.maxHeart != nil else {
+            return nil
+        }
+        
+        let base = [avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var paceRow: Int? {
+        guard self.pace != nil else {
+            return nil
+        }
+        
+        let base = [maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var speedRow: Int? {
+        guard self.speed != nil else {
+            return nil
+        }
+        
+        let base = [paceRow, maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var energyRow: Int? {
+        guard self.totalEnergy != nil else {
+            return nil
+        }
+        
+        let base = [speedRow, paceRow, maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    private var elevationRow: Int? {
+        let (asc, desc) = self.elevationChange
+        guard asc != nil || desc != nil else {
+            return nil
+        }
+        
+        let base = [energyRow, speedRow, paceRow, maxHeartRow, avgHeartRow, distanceRow].lazy.compactMap { $0 }.first ?? durationRow
+        return 1 + base
+    }
+    
+    public func ICSexport(for systemOfUnits: SystemOfUnits, _ callback: @escaping (URL?) -> Void) {
+        guard isLoaded, !hasError,
+            EKEventStore.authorizationStatus(for: EKEntityType.event) != EKAuthorizationStatus.denied,
+            EKEventStore.authorizationStatus(for: EKEntityType.event) != EKAuthorizationStatus.restricted
+        else {
+            callback(nil)
```

Comment:
This is not a real failure, we should prompt the user to update the authorization for calendar access